hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -124,6 +126,10 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	if err != nil {
 		r.Log.Error(err, "Failed to reconcile nodePool")
 		r.recorder.Eventf(nodePool, corev1.EventTypeWarning, "ReconcileError", "%v", err)
+		if err := patchHelper.Patch(ctx, nodePool); err != nil {
```

Comment:
Have you observed this working as expected? Many times now I've seen this utility succeed but not persist anything

Reply:
yeh, I tested this manually